kore/src/Kore/JsonRpc.hs

```diff
@@ -543,10 +544,11 @@ srv serverState moduleName Log.LoggerEnv{logAction} runSMT = do
 
     cancelError = ErrorObj "Request cancelled" (-32000) Null
 
-    bracketOnReqException before onCancel thing =
+    bracketOnReqException before onError thing =
         mask $ \restore -> do
             a <- before
-            restore (thing a) `catch` \(_ :: ReqException) -> onCancel a
+            (restore (thing a) `catch` \(_ :: ReqException) -> onError cancelError a)
+                `catch` \(err :: SomeException) -> print err >> onError (serverError "crashed" $ toJSON $ show err) a
```

Comment:
If we catch all exceptions here we won't need the `withErrHandler` inside `respond` any more.